### PR TITLE
feat(clouddriver): make restoreMin only restore the min size

### PR DESCRIPTION
**Pull request by dreynaud**
Inheriting the max and desired is a hazard that could lead to incidents.

### CODE REVIEW

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/aws/ApplySourceServerGroupCapacityTask.groovy [hunk 1]
```diff
@@ -67,10 +67,10 @@ class ApplySourceServerGroupCapacityTask extends AbstractServerGroupTask {
...
       )
 
-      // only update the min capacity, desired + max should be inherited/unchanged from the target server group
-      context.capacity = targetServerGroup.capacity + [
-        min: minCapacity
-      ]
+      // only update the min capacity
+      // this is important, as passing a stale 'desired' value here could interfere with autoscaling
+      // and result in instances that are needed being destroyed
+      context.capacity = [ min: minCapacity ]
```
[the remark this answers is not in the document]
Comment (ajordens) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/aws/ApplySourceServerGroupCapacityTask.groovy]: Maybe a little clearer?

```
// updating anything other than 'min' could result in instances being
// unnecessarily destroyed or created if autoscaling has occurred
```

**Comment by ajordens** [comment]
Thoughts @robzienert @robfletcher on whether we can have a Task/Stage change packages? I don't think we can but I know we were futzing with serialization.

**Comment by robzienert** [comment]
@ajordens no, we can't right now. Although if Daniel wants to do that, I can help guide. It's the last remaining vestige of old serialization for orca.

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/aws/ApplySourceServerGroupCapacityTask.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (dreynaud) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/aws/ApplySourceServerGroupCapacityTask.groovy]: Thanks for the suggestion, I missed that. Change in aaab6071a

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/Stage.java [hunk 2]
```diff
@@ -435,7 +436,7 @@ public Task taskById(String taskId) {
     try {
       return objectMapper.readValue(data, type);
     } catch (IOException e) {
-      throw new RuntimeException("Could not convert " + new String(data) + " to " + type.getSimpleName());
+      throw new RuntimeException("Could not convert " + new String(data, UTF_8) + " to " + type.getSimpleName(), e);
```
Comment (ajordens) [inline orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/Stage.java]: Did you see something that prompted this?

Comment (dreynaud) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/Stage.java]: noticed a compilation warning as I was testing, so I figured I'd fix

**Pull request merged**